nautobot/core/graphql/schema_init.py

```diff
@@ -1,14 +1,23 @@
 import graphene
 from graphene_django.types import ObjectType
+from django.db.models import signals
 
 from .schema import generate_query_mixin
+from nautobot.extras.models import CustomField, Relationship
 
 
-DynamicGraphQL = generate_query_mixin()
+def generate_schema(*args, **kwargs):
+    DynamicGraphQL = generate_query_mixin()
 
+    class Query(ObjectType, DynamicGraphQL):
+        """Contains the entire GraphQL Schema definition for Nautobot."""
 
-class Query(ObjectType, DynamicGraphQL):
-    """Contains the entire GraphQL Schema definition for Nautobot."""
+    return graphene.Schema(query=Query, auto_camelcase=False)
 
 
-schema = graphene.Schema(query=Query, auto_camelcase=False)
+schema = generate_schema()
+
+signals.post_save.connect(generate_schema, sender=Relationship)
+signals.post_delete.connect(generate_schema, sender=Relationship)
+signals.post_save.connect(generate_schema, sender=CustomField)
+signals.post_delete.connect(generate_schema, sender=CustomField)
```

Comment:
I'll admit I don't quite understand how this works - so when a Relationship or CustomField is saved or deleted, we now call generate_schema(), which generates a new schema and returns it.... to where, to be handled how?

Reply:
Don't we need signals for the Computed Fields as well?

Reply:
Yes we do